PWGCF/EbyEFluctuations/Tasks/netchargeFluctuations.cxx

```diff
@@ -257,8 +262,75 @@ struct NetchargeFluctuations {
     if (cItsTpcVtx && !coll.selection_bit(aod::evsel::kIsVertexITSTPC)) {
       return false;
     }
+    histogramRegistry.fill(HIST("hVtxZ_after"), coll.posZ());
+    histogramRegistry.fill(HIST("hCentrality"), cent);
+    histogramRegistry.fill(HIST("hMultiplicity"), mult);
 
-    return true; // if all checks pass, accept the collision
+    return true;
+  }
+  template <typename T>
+  void fillBeforeQA(T const& track)
+  {
+    histogramRegistry.fill(HIST("hTPCchi2perCluster_before"), track.tpcChi2NCl());
+    histogramRegistry.fill(HIST("hITSchi2perCluster_before"), track.itsChi2NCl());
+    histogramRegistry.fill(HIST("hTPCCrossedrows_before"), track.tpcNClsCrossedRows());
+    histogramRegistry.fill(HIST("hDcaXY_before"), track.dcaXY());
+    histogramRegistry.fill(HIST("hDcaZ_before"), track.dcaZ());
+    histogramRegistry.fill(HIST("hPtDcaXY_before"), track.pt(), track.dcaXY());
+    histogramRegistry.fill(HIST("hPtDcaZ_before"), track.pt(), track.dcaZ());
+  }
+
+  template <typename T>
+  void fillAfterQA(T const& track)
+  {
+    histogramRegistry.fill(HIST("hDcaXY_after"), track.dcaXY());
+    histogramRegistry.fill(HIST("hDcaZ_after"), track.dcaZ());
+    histogramRegistry.fill(HIST("hPt"), track.pt());
+    histogramRegistry.fill(HIST("hEta"), track.eta());
+    histogramRegistry.fill(HIST("hPtDcaXY_after"), track.pt(), track.dcaXY());
+    histogramRegistry.fill(HIST("hPtDcaZ_after"), track.pt(), track.dcaZ());
+    histogramRegistry.fill(HIST("hTPCCrossedrows_after"), track.tpcNClsCrossedRows());
+    histogramRegistry.fill(HIST("hTPCchi2perCluster_after"), track.tpcChi2NCl());
+    histogramRegistry.fill(HIST("hITSchi2perCluster_after"), track.itsChi2NCl());
+  }
+
+  template <typename N, typename C>
+  void nchDistribution(N const& nch, C const& cent)
```

Comment:
Why is needed to template on the type of `nch` and of `cent`? Aren't they clearly know at coding time?